ydbd_slice: add support for using raw config.yaml

melazyk opened the pull request:
ydbd_slice: add support for using raw config.yaml via --yaml-config and dynconfig.yaml via --yaml-dynconfig



### Changelog entry 

- Add yaml_configurator instead of configurator
- Add warning for old style cluster.yaml
- Add dynconfig-generator for generating simple dynconfig.yaml
- Add --save-raw-cfg option for saving raw config from old cluster.yaml

### Changelog category 

* Experimental feature


### Description for reviewers
This PR introduces a new slice with the following changes:

1. **Configuration Files**:
   - Created `config.yaml` and `cluster.yaml` (containing only the `domains` section).
   - Generated the minimally required `dynconfig.yaml` using:
     ```bash
     ydbd_slice dynconfig-generator --yaml-config config.yaml --output-file dynconfig.yaml
     ```

2. **Slice Creation**:
   - The slice is created using the following command:
     ```bash
     ydbd_slice install cluster.yaml all --yaml-config config.yaml --binary /path/to/ydbd
     ```

3. **Notes**:
   - If `ydbd_slice` is run without the `--yaml-config` parameter, a warning with instructions will be displayed.

melazyk committed:
ydbd_slice: add support for using raw config.yaml via --yaml-config and dynconfig.yaml via --yaml-dynconfig

- Add yaml_configurator instead of configurator
- Add warning for old style cluster.yaml
- Add dynconfig-generator for generating simple dynconfig.yaml
- Add --save-raw-cfg option for saving raw config from old cluster.yaml

kobzonega commented on ydb/tools/ydbd_slice/yaml_configurator.py:
```diff
@@ -0,0 +1,169 @@
+        return [host['host'] for host in self.static_dict.get('hosts', [])]
+
+    @property
+    def kickimr_cfg(self):
```
опечатка? 
```suggestion
    def kikimr_cfg(self):
```

melazyk replied:
опечатка, не влияет на функционал, но поправил, чтобы было верно по смыслу

kobzonega commented on ydb/tools/ydbd_slice/yaml_configurator.py:
```diff
@@ -0,0 +1,169 @@
+                'cluster': cluster_uuid,
+                'version': 0,
+            },
+            'config': copy.deepcopy(self.__yaml_config),
```
А давай удалим вот эти секции из статического конфига при переносе в динконфиг? (как это сделано в ydb-kuebrnetes-operator)
Они там правда не нужны в динконфиге, это все спецификация storage
https://github.com/ydb-platform/ydb-kubernetes-operator/blob/master/api/v1alpha1/configuration.go#L141-L149

melazyk replied:
убрал перечисленные ключи при генерации dynconfig.yaml

melazyk committed:
ydbd_slice: review fixes

- --yaml-dynconfig removed
- dynconfig.yaml generator fixed ( remove storage only options )
- fix the typo

melazyk committed:
ydbd_slice:  yaml_configurator.py flake8 test fix
